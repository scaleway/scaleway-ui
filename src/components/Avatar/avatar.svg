<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 104 104">
  <rect width="103.88" height="103.88" rx="21.85" fill="#ba00ff" />
  <path
    d="M84.85 69.41a34.43 34.43 0 003.72-14.1c-1.37 4.52-5.78 7.94-9.91 9.83a27.21 27.21 0 01-14.72 2A31.07 31.07 0 0149.74 62c-4-2.92-6.17-7.38-7.34-12.12a23.33 23.33 0 01.74-14.63 32 32 0 014.15-6.64 16.64 16.64 0 015.46-4.86 25.36 25.36 0 016.91-2.93 15.57 15.57 0 016 0 37.61 37.61 0 00-13.69-2.57h-.56a28.59 28.59 0 00-15.1 11.08A34.88 34.88 0 0031 54.57a40.9 40.9 0 004.25 12.86A19.45 19.45 0 0044.6 76c8 3.66 18.07 4.61 26.47 1.79a24.07 24.07 0 0014.15-12.22c.39-1.31.75-2.63 1.11-3.94a.42.42 0 01.82.22c-.69 2.52-1.41 5.08-2.3 7.56z"
    fill="#51069d" />
  <path
    d="M59.66 20.83a25.36 25.36 0 00-6.91 2.93 16.64 16.64 0 00-5.46 4.86 32 32 0 00-4.15 6.64 23.33 23.33 0 00-.74 14.63c1.17 4.74 3.31 9.2 7.34 12.12a31.07 31.07 0 0014.2 5.17 27.21 27.21 0 0014.72-2c4.13-1.89 8.54-5.31 9.91-9.83v-1.54a35.5 35.5 0 00-23-32.91 15.57 15.57 0 00-5.91-.07z"
    fill="#672cbc" />
  <path
    d="M56.81 110.15a.17.17 0 01.32.06c.14 1.67.13 3.35.17 5a17.36 17.36 0 002.38 0 15 15 0 002.46-.34 74.36 74.36 0 003.22-11.2c-6.06 0-11.12 5.62-11.12 5.62l.16 1.41a4.3 4.3 0 012.37-.6z"
    fill="#a990fc" />
  <path
    d="M41.5 114.91a14.78 14.78 0 002.3.33c-.19-.61-.37-1.23-.51-1.86-.48-2.29-.59-4.65-1-6.95a12.64 12.64 0 00-.59-2.16 11.19 11.19 0 00-3.38-.56 74.36 74.36 0 003.18 11.2z"
    fill="#672cbc" />
  <path
    d="M74.43 102.06a15.43 15.43 0 00-.26-2.56 13.39 13.39 0 01-3.34 1.24c.09.56.16 1.13.21 1.69a6.52 6.52 0 01-.22 3 22.84 22.84 0 01-1.56 2.66 10.41 10.41 0 00-.63 1.49c1.12-.45 3-3.7 3-3.7s-1.86 7.07-.5 7.66c.47.2 1-.44 1.44-1.44 1.65-2.54 1.95-5.6 1.86-10.04z"
    fill="#51069d" />
  <path
    d="M67.17 89.74a2.85 2.85 0 00.54 2.73 25.21 25.21 0 011.6 3.1 22.59 22.59 0 011.52 5.17 13.39 13.39 0 003.34-1.24 25.26 25.26 0 00-5.44-10.85 2.55 2.55 0 00-1.56 1.09z"
    fill="#a990fc" />
  <path
    d="M70.82 105.44a6.52 6.52 0 00.22-3c-.05-.56-.12-1.13-.21-1.69a12.93 12.93 0 01-1.76.29c.45 1.2.78 2.23 1 2.91 0 0-2.86 4.75-1.95 5.6a.45.45 0 00.51 0 10.41 10.41 0 01.63-1.49 22.84 22.84 0 001.56-2.62z"
    fill="#350072" />
  <path
    d="M65 93.83a22.89 22.89 0 014.1 7.2 12.93 12.93 0 001.76-.29 22.59 22.59 0 00-1.52-5.17 25.21 25.21 0 00-1.6-3.1 2.85 2.85 0 01-.54-2.73 2.55 2.55 0 011.56-1.09c-.55-.65-1-1.21-1.45-1.63a2.79 2.79 0 00-.34-.28 15.48 15.48 0 00-3.29 1.08c-.85 1.4-.85 3.65 1.32 6.01z"
    fill="#672cbc" />
  <path
    d="M30.58 100.29a21 21 0 00-.88 3.84 17.15 17.15 0 00.44 7.6c.07.13.14.26.22.38.44 1 1 1.64 1.44 1.44 1.37-.59-.49-7.66-.49-7.66s2.6 4.5 3.51 3.65-2-5.6-2-5.6c.23-.68.55-1.71 1-2.91a13.56 13.56 0 01-3.24-.74z"
    fill="#51069d" />
  <path
    d="M37.42 91.46c-1.28-.75-1.53-.07-2.11.42a15.22 15.22 0 00-2.66 3.74 24.33 24.33 0 00-2.07 4.67 13.56 13.56 0 003.29.74 23.09 23.09 0 014.1-7.2 7.91 7.91 0 001-1.31 2.27 2.27 0 01-.75-.32c-.31-.2-.51-.57-.8-.74z"
    fill="#a990fc" />
  <path d="M30.58 100.29a13.56 13.56 0 003.29.74 13.56 13.56 0 01-3.29-.74z" fill="#a990fc" />
  <path
    d="M29.7 104.13a21 21 0 01.88-3.84 14 14 0 01-1.8-.79 15.43 15.43 0 00-.27 2.56c-.08 4.22.19 7.2 1.63 9.67a17.15 17.15 0 01-.44-7.6z"
    fill="#350072" />
  <path
    d="M35.67 87c-1.93 2-5.84 6.79-6.89 12.48a14 14 0 001.8.79 24.33 24.33 0 012.07-4.67 15.22 15.22 0 012.66-3.74c.58-.49.83-1.17 2.11-.42.29.17.49.54.79.74a2.27 2.27 0 00.75.32c2.54-4.19-1.28-7.5-3.29-5.5z"
    fill="#672cbc" />
  <path d="M30.58 100.29a14 14 0 01-1.8-.79 14 14 0 001.8.79z" fill="#350072" />
  <path d="M51.82 82.53h0z" fill="none" stroke="#000" stroke-miterlimit="10" stroke-width=".353" />
  <path
    d="M44.47 88.35c-.69-.09-1.38-.2-2.07-.32a38 38 0 009.54 1.22 37.19 37.19 0 005-.34 63.11 63.11 0 01-12.47-.56z"
    fill="#51069d" />
  <path
    d="M44.39 100.43c-.54 0-1-.06-1.52-.11.48.05 1 .09 1.52.11 0 0 .65 6 7.49 6-6.88.05-7.49-6-7.49-6z"
    fill="none" />
  <path
    d="M65.23 104.3a13 13 0 001-5.45 15.89 15.89 0 01-6.83 1.58s-.65 6-7.48 6c-6.84 0-7.49-6-7.49-6-.54 0-1-.06-1.52-.11 0 .21.07.41.11.62a9.84 9.84 0 004.57 6.82 15.59 15.59 0 0015.55-.56 10.18 10.18 0 002.09-2.9zM51.8 92a51.1 51.1 0 01-8.57-.27 17.76 17.76 0 00-.49 4.13 24.65 24.65 0 000 2.72 18.61 18.61 0 001.89.1s.43-6.14 7.35-6.14 7.35 6.14 7.35 6.14a16.17 16.17 0 006.85-1.41 46.55 46.55 0 00-1-6.92 6.91 6.91 0 01-1.25.26A46.88 46.88 0 0151.8 92zm5.08-3.6c1.24 0 2.5-.1 3.76-.14a11.84 11.84 0 013.49-.55h.38c-.33-1.26-.58-2-.58-2s-3.59-2.62-12-3.12c-8.47.5-12.06 3.12-12.06 3.12s-.19.61-.46 1.62c3.79.42 7.52 1 11.35 1.11 2.01 0 4.06-.01 6.12-.04z"
    fill="#672cbc" />
  <path
    d="M51.94 92.51c-6.92 0-7.35 6.14-7.35 6.14a18.61 18.61 0 01-1.89-.1 17.38 17.38 0 00.17 1.77c.48.05 1 .09 1.52.11 0 0 .65 6 7.49 6s7.48-6 7.48-6a15.89 15.89 0 006.83-1.58v-1.61a16.17 16.17 0 01-6.85 1.41s-.48-6.14-7.4-6.14z"
    fill="#51069d" />
  <path
    d="M59.36 100.43s-.65 6-7.48 6c6.83 0 7.48-6 7.48-6a15.89 15.89 0 006.83-1.58 15.89 15.89 0 01-6.83 1.58z"
    fill="#51069d" />
  <path
    d="M47.55 107.76a9.84 9.84 0 01-4.55-6.82c0-.21-.07-.41-.11-.62a14.93 14.93 0 01-5.31-1.47 12.83 12.83 0 001 5.45c2.75 5.72 10.61 7.3 13.35 7.32 2.2 0 7.71-1 11.22-4.42a15.59 15.59 0 01-15.6.56zm-4.85-9.21a24.65 24.65 0 010-2.72 17.76 17.76 0 01.49-4.13 51.1 51.1 0 008.61.3 46.88 46.88 0 0012.08-1.39 6.91 6.91 0 001.25-.26c-.21-1-.43-1.89-.62-2.62h-.38a11.84 11.84 0 00-3.49.55c-1.26 0-2.52.12-3.76.14h-6.17c-3.83-.1-7.56-.69-11.35-1.11a53.36 53.36 0 00-1.75 9.91 15.36 15.36 0 005.09 1.33z"
    fill="#51069d" />
  <path
    d="M37.61 97.18v1.67a14.93 14.93 0 005.31 1.47 17.38 17.38 0 01-.17-1.77 15.36 15.36 0 01-5.14-1.37zm47.24-27.77c.89-2.48 1.61-5 2.3-7.56a.42.42 0 00-.82-.22c-.36 1.31-.72 2.63-1.11 3.94a24.07 24.07 0 01-14.15 12.21C62.67 80.6 52.61 79.65 44.6 76a19.45 19.45 0 01-9.37-8.56A40.9 40.9 0 0131 54.57a34.88 34.88 0 015.3-25.18 28.59 28.59 0 0115.1-11.08c-20 .28-36.1 16.05-36.1 35.46C15.28 70.17 26.78 84 42.4 88c.69.12 1.38.23 2.07.32a63.11 63.11 0 0012.45.56 36.65 36.65 0 0027.93-19.47z"
    fill="#350072" />
  <path
    d="M67.22 32c-2.5 1.61-1.72 3 1.4 5.94 4.71 4.49 2.38 11.67 7.68 10.45s2.19-9 .22-11.57-6-6.92-9.3-4.78M80 53.56a3.34 3.34 0 10-2.88 3.73A3.34 3.34 0 0080 53.56"
    fill="#fff" />
  <path
    d="M89.33 47c-2.15 0-4.59 1.18-4.59 8.4 0 29.87-24.21 30.46-32.8 30.46s-32.8-.59-32.8-30.46c0-7.22-2.44-8.4-4.59-8.4-1 0-2.39.86-3.27 2.63a3.79 3.79 0 014 .15c3.29 2.32 1.88 7 1.89 10.32 0 7.2 3.63 14 8.86 18.86 6 5.59 13.92 8.26 22.07 8.57a61.82 61.82 0 0024.6-3.73 41.14 41.14 0 0010.18-5.71c6.55-7.2 9.05-15.63 10.06-19.67C94.9 50.61 91.48 47 89.33 47z"
    fill="#fff" />
  <path
    d="M48.07 87.52c-8.15-.31-16-3-22.07-8.57-5.23-4.82-8.85-11.66-8.86-18.86 0-3.33 1.4-8-1.89-10.32a3.79 3.79 0 00-4-.15c-.93 1.88-1.35 4.78-.34 8.8 2 7.81 9.47 32 41 32 15.21 0 24.84-5.64 30.94-12.35A41.14 41.14 0 0172.7 83.8a61.82 61.82 0 01-24.63 3.72z"
    fill="#a990fc" />
  <path
    d="M92.4 56.66c.45-3.2-.14-5.93-1.31-6.09s-2.49 2.29-2.94 5.5.14 5.93 1.31 6.1 2.49-2.3 2.94-5.51m-80.89 0c-.45-3.2.14-5.93 1.32-6.09s2.49 2.29 2.94 5.5-.14 5.93-1.32 6.1-2.49-2.3-2.94-5.51"
    fill="#fe6776" />
  <path
    d="M12.46 56.49c-.19-1.34.06-2.49.55-2.56s1.05 1 1.23 2.31-.05 2.49-.55 2.56-1-1-1.23-2.31m78.99.04c.19-1.34 0-2.49-.55-2.56s-1 1-1.23 2.31.06 2.49.55 2.56 1-1 1.23-2.31M57.11 99.31a5.3 5.3 0 10-5.29 5.3 5.29 5.29 0 005.29-5.3"
    fill="#fff" />
  <path d="M55.91 99.31a4 4 0 10-4 4 4 4 0 004-4" fill="#fe6776" />
  <path d="M63.65 87.82a15.48 15.48 0 013.29-1.08 2.34 2.34 0 00-3.29 1.08z" fill="#a990fc" />
  <rect width="103.88" height="103.88" rx="21.85" fill="none" />
</svg>
