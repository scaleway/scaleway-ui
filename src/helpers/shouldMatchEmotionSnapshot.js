/* eslint-disable import/no-extraneous-dependencies */
import createCache from '@emotion/cache'
import { createSerializer } from '@emotion/jest'
import { CacheProvider, ThemeProvider } from '@emotion/react'
import { render } from '@testing-library/react'
import React from 'react'
import theme from '../theme'

expect.addSnapshotSerializer(createSerializer())

const emotionCache = createCache({
  key: 'cache',
})
emotionCache.compat = true

export default async (component, { transform } = {}) => {
  const node = render(
    <CacheProvider value={emotionCache}>
      <ThemeProvider theme={theme}>{component}</ThemeProvider>
    </CacheProvider>,
  )
  if (transform) await transform(node)

  expect(node.asFragment()).toMatchSnapshot()
}
