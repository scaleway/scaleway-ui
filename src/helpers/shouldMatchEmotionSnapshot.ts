/* eslint-disable import/no-extraneous-dependencies */
import { createSerializer } from '@emotion/jest'
import { RenderOptions, render } from '@testing-library/react'
import { ReactNode } from 'react'
import defaultTheme from '../theme'
import renderWithTheme from './renderWithTheme'

// use only class hash (generated from css style content)
expect.addSnapshotSerializer(
  createSerializer({ classNameReplacer: className => className }),
)

interface Options {
  options?: RenderOptions
  theme?: typeof defaultTheme
  transform?: (node: ReturnType<typeof render>) => Promise<void> | void
}

export default async (
  component: ReactNode,
  { transform, options, theme }: Options = {},
): Promise<void> => {
  const node = renderWithTheme(component, options, theme)
  if (transform) await transform(node)

  expect(node.asFragment()).toMatchSnapshot()
  node.unmount()
}
