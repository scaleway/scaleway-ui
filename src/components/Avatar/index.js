import styled from '@emotion/styled'
import PropTypes from 'prop-types'
import React from 'react'
import { Box } from '../Box'
import Icon from '../Icon'
import avatar from './avatar.svg'

const formatTextToAvatar = text => {
  if (text.length <= 2) {
    return text.toUpperCase()
  }

  if (text.split(' ').length > 1) {
    const [a, b] = text.split(' ')

    return `${a[0]}${b[0]}`.toUpperCase()
  }

  return text.substring(0, 2).toUpperCase()
}

const StyledDiv = styled.div`
  align-items: center;
  background-color: ${({ lock, theme, textBgColor }) =>
    lock ? theme.colors.gray50 : theme.colors[textBgColor] || textBgColor};
  border-radius: 50%;
  color: ${({ theme, textColor }) => theme.colors[textColor] || textColor};
  font-size: ${({ textSize }) => textSize}px;
  display: flex;
  height: 100%;
  justify-content: center;
  width: 100%;
`

const StyledImg = styled.img`
  height: 100%;
  width: 100%;
`

const Avatar = ({
  image,
  size,
  text,
  textBgColor,
  textColor,
  textSize,
  lock,
  ...props
}) => (
  <Box width={size} height={size} position="relative" {...props}>
    {text ? (
      <StyledDiv
        lock={lock}
        textBgColor={textBgColor}
        textColor={textColor}
        textSize={textSize}
      >
        {lock ? <Icon name="lock" color="gray550" /> : formatTextToAvatar(text)}
      </StyledDiv>
    ) : (
      <StyledImg src={image} alt="" />
    )}
  </Box>
)

Avatar.defaultProps = {
  image: avatar,
  size: 32,
  text: null,
  textBgColor: 'lightViolet',
  textColor: 'white',
  textSize: 10,
  lock: false,
}

Avatar.propTypes = {
  image: PropTypes.string,
  size: PropTypes.number,
  text: PropTypes.string,
  textBgColor: PropTypes.string,
  textColor: PropTypes.string,
  textSize: PropTypes.number,
  lock: PropTypes.bool,
}

export default Avatar
