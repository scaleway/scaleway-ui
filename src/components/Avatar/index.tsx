import styled from '@emotion/styled'
import PropTypes from 'prop-types'
import React, { FunctionComponent } from 'react'
import { ColorDeprecated as Color } from '../../theme/deprecated/colors'
import Box from '../Box'
import Icon from '../Icon'
import avatar from './avatar.svg'

const formatTextToAvatar = (text: string): string => {
  if (text.length <= 2) {
    return text.toUpperCase()
  }

  if (text.split(' ').length > 1) {
    const [a, b] = text.split(' ')

    return `${a[0]}${b[0]}`.toUpperCase()
  }

  return text.substring(0, 2).toUpperCase()
}

const StyledDiv = styled.div<{
  lock?: boolean
  textBgColor?: string
  textColor: string
  textSize: number
}>`
  align-items: center;
  background-color: ${({ lock, theme, textBgColor }) =>
    lock
      ? theme.colorsDeprecated.gray50
      : theme.colorsDeprecated[textBgColor as Color] || textBgColor};
  border-radius: 50%;
  color: ${({ theme, textColor }) =>
    theme.colorsDeprecated[textColor as Color] || textColor};
  font-size: ${({ textSize }) => textSize}px;
  display: flex;
  height: 100%;
  justify-content: center;
  width: 100%;
`

const StyledImg = styled.img`
  height: 100%;
  width: 100%;
`

interface AvatarProps {
  image?: string
  size?: number
  text?: string
  textBgColor?: string
  textColor?: string
  textSize?: number
  lock?: boolean
}

const Avatar: FunctionComponent<AvatarProps> = ({
  image = avatar,
  size = 32,
  text,
  textBgColor = 'lightViolet',
  textColor = 'white',
  textSize = 10,
  lock = false,
  ...props
}) => (
  <Box width={size} height={size} position="relative" {...props}>
    {text ? (
      <StyledDiv
        lock={lock}
        textBgColor={textBgColor}
        textColor={textColor}
        textSize={textSize}
      >
        {lock ? <Icon name="lock" color="gray550" /> : formatTextToAvatar(text)}
      </StyledDiv>
    ) : (
      <StyledImg src={image} alt="" />
    )}
  </Box>
)

Avatar.propTypes = {
  image: PropTypes.string,
  /**
   * Used only when `text` prop is specified
   */
  lock: PropTypes.bool,
  /**
   * Size of the component
   */
  size: PropTypes.number,
  text: PropTypes.string,
  /**
   * Background color when `text` prop is specified
   */
  textBgColor: PropTypes.string,
  /**
   * Text color when `text` prop is specified
   */
  textColor: PropTypes.string,
  /**
   * Text size when `text` prop is specified or size of the lock when `lock` is true
   */
  textSize: PropTypes.number,
}

export default Avatar
