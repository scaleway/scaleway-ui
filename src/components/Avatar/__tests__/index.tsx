import React from 'react'
import Avatar from '..'
import shouldMatchEmotionSnapshot from '../../../helpers/shouldMatchEmotionSnapshot'

describe('Avatar', () => {
  it('renders correctly with default props', () =>
    shouldMatchEmotionSnapshot(<Avatar />))

  it('renders correctly with sentence', () =>
    shouldMatchEmotionSnapshot(<Avatar text="Hello World" />))

  it('renders correctly with lock', () =>
    shouldMatchEmotionSnapshot(<Avatar text="HelloWorld" lock />))

  it('renders correctly with text', () =>
    shouldMatchEmotionSnapshot(<Avatar text="HelloWorld" />))

  it('renders correctly with small text', () =>
    shouldMatchEmotionSnapshot(<Avatar text="HW" />))

  it('renders correctly with text and custom style', () =>
    shouldMatchEmotionSnapshot(
      <Avatar text="HW" textBgColor="red" textColor="white" textSize={14} />,
    ))

  it('renders correctly with custom text color', () =>
    shouldMatchEmotionSnapshot(<Avatar text="HW" textColor="#FFFFFF" />))

  it('renders correctly with custom background color', () =>
    shouldMatchEmotionSnapshot(<Avatar text="HW" textBgColor="#FFFFFF" />))
})
