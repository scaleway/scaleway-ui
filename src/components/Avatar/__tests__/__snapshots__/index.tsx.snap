// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar renders correctly with custom background color 1`] = `
<DocumentFragment>
  .cache-hjx5aa-AvatarContainer {
  height: 32px;
  width: 32px;
}

.cache-1cjx2xd-StyledTextAvatar {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #FFFFFF;
  border-radius: 50%;
  color: #ffffff;
  font-size: 10px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  width: 100%;
}

<div
    class="cache-hjx5aa-AvatarContainer e1frqeh80"
    size="32"
  >
    <span
      class="cache-1cjx2xd-StyledTextAvatar e1frqeh82"
    >
      HW
    </span>
  </div>
</DocumentFragment>
`;

exports[`Avatar renders correctly with custom text color 1`] = `
<DocumentFragment>
  .cache-hjx5aa-AvatarContainer {
  height: 32px;
  width: 32px;
}

.cache-gstjrq-StyledTextAvatar {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #a365f6;
  border-radius: 50%;
  color: #FFFFFF;
  font-size: 10px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  width: 100%;
}

<div
    class="cache-hjx5aa-AvatarContainer e1frqeh80"
    size="32"
  >
    <span
      class="cache-gstjrq-StyledTextAvatar e1frqeh82"
    >
      HW
    </span>
  </div>
</DocumentFragment>
`;

exports[`Avatar renders correctly with default props 1`] = `
<DocumentFragment>
  .cache-hjx5aa-AvatarContainer {
  height: 32px;
  width: 32px;
}

.cache-fpgk3f-StyledImg {
  height: 100%;
  width: 100%;
}

<div
    class="cache-hjx5aa-AvatarContainer e1frqeh80"
    size="32"
  >
    <img
      alt=""
      class="cache-fpgk3f-StyledImg e1frqeh81"
      src="test-file-stub"
    />
  </div>
</DocumentFragment>
`;

exports[`Avatar renders correctly with lock 1`] = `
<DocumentFragment>
  .cache-hjx5aa-AvatarContainer {
  height: 32px;
  width: 32px;
}

.cache-e89vgj-StyledTextAvatar {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #f6f6f8;
  border-radius: 50%;
  color: #ffffff;
  font-size: 10px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  width: 100%;
}

.cache-67zp1h-StyledIcon-sizeStyles {
  fill: #b2b6c3;
  height: 1em;
  width: 1em;
  min-width: 1em;
  min-height: 1em;
}

.cache-67zp1h-StyledIcon-sizeStyles.cache-67zp1h-StyledIcon-sizeStyles {
  vertical-align: middle;
}

<div
    class="cache-hjx5aa-AvatarContainer e1frqeh80"
    size="32"
  >
    <span
      class="cache-e89vgj-StyledTextAvatar e1frqeh82"
    >
      <svg
        class="sc-ui-icon etwatq50 cache-67zp1h-StyledIcon-sizeStyles"
        title="Locked"
        viewBox="0 0 24 24"
      >
        <path
          d="M12,17A2,2 0 0,0 14,15C14,13.89 13.1,13 12,13A2,2 0 0,0 10,15A2,2 0 0,0 12,17M18,8A2,2 0 0,1 20,10V20A2,2 0 0,1 18,22H6A2,2 0 0,1 4,20V10C4,8.89 4.9,8 6,8H7V6A5,5 0 0,1 12,1A5,5 0 0,1 17,6V8H18M12,3A3,3 0 0,0 9,6V8H15V6A3,3 0 0,0 12,3Z"
        />
      </svg>
    </span>
  </div>
</DocumentFragment>
`;

exports[`Avatar renders correctly with sentence 1`] = `
<DocumentFragment>
  .cache-hjx5aa-AvatarContainer {
  height: 32px;
  width: 32px;
}

.cache-1dabazh-StyledTextAvatar {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #a365f6;
  border-radius: 50%;
  color: #ffffff;
  font-size: 10px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  width: 100%;
}

<div
    class="cache-hjx5aa-AvatarContainer e1frqeh80"
    size="32"
  >
    <span
      class="cache-1dabazh-StyledTextAvatar e1frqeh82"
    >
      HW
    </span>
  </div>
</DocumentFragment>
`;

exports[`Avatar renders correctly with size 1`] = `
<DocumentFragment>
  .cache-xgtog4-AvatarContainer {
  height: 48px;
  width: 48px;
}

.cache-fpgk3f-StyledImg {
  height: 100%;
  width: 100%;
}

<div
    class="cache-xgtog4-AvatarContainer e1frqeh80"
    size="48"
  >
    <img
      alt=""
      class="cache-fpgk3f-StyledImg e1frqeh81"
      src="test-file-stub"
    />
  </div>
</DocumentFragment>
`;

exports[`Avatar renders correctly with small text 1`] = `
<DocumentFragment>
  .cache-hjx5aa-AvatarContainer {
  height: 32px;
  width: 32px;
}

.cache-1dabazh-StyledTextAvatar {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #a365f6;
  border-radius: 50%;
  color: #ffffff;
  font-size: 10px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  width: 100%;
}

<div
    class="cache-hjx5aa-AvatarContainer e1frqeh80"
    size="32"
  >
    <span
      class="cache-1dabazh-StyledTextAvatar e1frqeh82"
    >
      HW
    </span>
  </div>
</DocumentFragment>
`;

exports[`Avatar renders correctly with text 1`] = `
<DocumentFragment>
  .cache-hjx5aa-AvatarContainer {
  height: 32px;
  width: 32px;
}

.cache-1dabazh-StyledTextAvatar {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #a365f6;
  border-radius: 50%;
  color: #ffffff;
  font-size: 10px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  width: 100%;
}

<div
    class="cache-hjx5aa-AvatarContainer e1frqeh80"
    size="32"
  >
    <span
      class="cache-1dabazh-StyledTextAvatar e1frqeh82"
    >
      HE
    </span>
  </div>
</DocumentFragment>
`;

exports[`Avatar renders correctly with text and custom style 1`] = `
<DocumentFragment>
  .cache-hjx5aa-AvatarContainer {
  height: 32px;
  width: 32px;
}

.cache-1dmd3xz-StyledTextAvatar {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: red;
  border-radius: 50%;
  color: white;
  font-size: 14px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  width: 100%;
}

<div
    class="cache-hjx5aa-AvatarContainer e1frqeh80"
    size="32"
  >
    <span
      class="cache-1dmd3xz-StyledTextAvatar e1frqeh82"
    >
      HW
    </span>
  </div>
</DocumentFragment>
`;

exports[`Avatar renders correctly with text size 1`] = `
<DocumentFragment>
  .cache-1yyml98-AvatarContainer {
  height: 50px;
  width: 50px;
}

.cache-qdcgyh-StyledTextAvatar {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #a365f6;
  border-radius: 50%;
  color: #ffffff;
  font-size: 30px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  width: 100%;
}

<div
    class="cache-1yyml98-AvatarContainer e1frqeh80"
    size="50"
  >
    <span
      class="cache-qdcgyh-StyledTextAvatar e1frqeh82"
    >
      HE
    </span>
  </div>
</DocumentFragment>
`;
